tests/foreman/ui/test_host.py

```diff
@@ -2451,3 +2465,46 @@ def test_positive_host_registration_with_non_admin_user(
 
         result = rhel8_contenthost.execute(cmd)
         assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+
+@pytest.mark.tier2
+def test_all_hosts_delete(session, target_sat, function_org, function_location, new_host_ui):
+    """Create a host and delete it through All Hosts UI
+
+    :id: 42b4560c-bb57-4c58-928e-e5fd5046b93f
+
+    :expectedresults: Successful deletion of a host through the table dropdown
+
+    :CaseComponent:Hosts-Content
+
+    :Team: Phoenix-content
+
+    :CaseLevel: System
+    """
+    host = target_sat.api.Host(organization=function_org, location=function_location).create()
+    with target_sat.ui_session() as session:
+        session.organization.select(function_org.name)
+        session.location.select(function_location.name)
+        assert session.all_hosts.delete(host.name)
+
```

Comment:
Additional negative assertion can be use to check host is being delete or not
```suggestion
assert not session.host.search(host.name)
```
OR below suggestion can be used.
```suggestion
assert not target_sat.api.Host().search(query={'search': f'name="{host.name}"'})
```

Reply:
Yea, `view.no_results` will give expected value for assertion.